### DISCUSSION TITLE
Exact properties of multi-isolate support

### DISCUSSION THREAD

**Discussion by tiloc** [isar/isar#20] [Help]
I would like to understand the planned abilities or limitations of Isar in a multi-isolate or multi-process environment. I am looking for a database that can successfully be used with https://pub.dev/packages/background_fetch or another package like the android_alarm_manager or workmanager. I want to run data fetches from a RESTful service in the background even when my main app is closed, write it into the database and listen / read the data in my main app (while the background service continues to execute periodically).
Hive was not suited for that (no support for multi-process). Sembast and Sqlflite also had limitations. What is the situation with Isar?

**Comment by simc** [comment]
In the immediate future, no multi process support is planned unfortunately. Multi isolate (=multi thread) support works out of the box and also supports collection, object and query notifiers.

**Comment by simc** [comment]
To be clear: There is nothing preventing multi process support. LMDB supports it and so could Isar. The only issue is creating some kind of IPC to notify watchers in other processes. This can definitely be done and I'd be happy to review PRs.

**Comment by LucaCoduriV** [comment]
Does it mean that it don't work with WorkManager ?
